### PR TITLE
[flink] Resolve bounded stream input finished in multiple checkpoint interval then cause commit failed

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Stop call end input if streaming checkpoint enabled. Because prepareCheckpointBarrier will handle the last checkpoint method.

Do as `CommitterOperator`

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
We need a test case.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java [hunk 1]
```diff
@@ -61,6 +64,10 @@ public void setup(
             StreamConfig config,
             Output<StreamRecord<OUT>> output) {
         super.setup(containingTask, config, output);
+        this.streamingCheckpointEnabled =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java:67]: extract a method for `streamingCheckpointEnabled `

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSink.java [hunk 2]
```diff
@@ -153,6 +153,9 @@ public SingleOutputStreamOperator<Committable> doWrite(
                 StreamExecutionEnvironmentUtils.getConfiguration(env)
                                 .get(ExecutionOptions.RUNTIME_MODE)
                         == RuntimeExecutionMode.STREAMING;
+        CheckpointConfig checkpointConfig = env.getCheckpointConfig();
+        boolean streamingCheckpointEnabled =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSink.java:157]: extract a method: `streamingCheckpointEnabled(env)`

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java [hunk 3]
```diff
@@ -48,6 +49,7 @@ public abstract class PrepareCommitOperator<IN, OUT> extends AbstractStreamOpera
     @Nullable protected transient MemorySegmentPool memoryPool;
     @Nullable private transient MemorySegmentAllocator memoryAllocator;
     private final Options options;
+    private boolean streamingCheckpointEnabled;
     private boolean endOfInput = false;
 
     public PrepareCommitOperator(Options options) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java:55]: PrepareCommitOperator(Options options, streamingCheckpointEnabled)

Comment (leaves12138) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java]: resolved

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/CommitterOperator.java [hunk 4]
```diff
@@ -193,16 +193,24 @@ private void pollInputs() throws Exception {
         for (Map.Entry<Long, List<CommitT>> entry : grouped.entrySet()) {
             Long cp = entry.getKey();
             List<CommitT> committables = entry.getValue();
-            if (committablesPerCheckpoint.containsKey(cp)) {
+            if (cp != null && cp == Long.MAX_VALUE && committablesPerCheckpoint.containsKey(cp)) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/CommitterOperator.java:196]: Add documentation here, why we need to do this

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**